testing/test_assertion.py

```diff
@@ -776,6 +776,17 @@ def test_mojibake(self) -> None:
         msg = "\n".join(expl)
         assert msg
 
+    def test_nfc_nfd_same_string(self) -> None:
+        # issue 3426
+        left = "hyv\xe4"
+        right = "hyva\u0308"
+        expl = callequal(left, right)
+        assert expl == [
+            "'hyvä' == 'hyvä'",
```

Comment:
This is the line I'd really like to change - I think we can do that by adding a `use_ascii: bool` flag to `saferepr()`, and then setting `use_ascii = isinstance(left, str) and isinstance(right, str) and nfd(left) == nfd(right)` in `assertrepr_compare()`.

At that point I think we won't need to change `_diff_text()` in this PR at all.

Reply:
Oh, I see. That makes sense.

However, now only the first line is indicative of where the strings are different, and we have the not-very-helpful:

```
    - hyva
    + hyva
```
as the diff_text. 

Are we happy with this, as long as the first line is clear?